File: app/src/main/java/com/example/lab8/DetailFragment.java
package com.example.lab8;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


public class DetailFragment extends Fragment {

    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;

    private String name;
    private String fullDescribtion;
    private String price;
    private String describtion;

    public DetailFragment() {

    }


    public static DetailFragment newInstance(String param1, String param2) {
        DetailFragment fragment = new DetailFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.detail_fragment, container, false);

        DetailFragmentArgs args =    DetailFragmentArgs.fromBundle(getArguments());
        fullDescribtion = args.getFullDescribtion();
        name = args.getName();
      describtion = args.getDescribtion();
     price = args.getPrice();
        TextView titleText = view.findViewById(R.id.textView);
        TextView describtionText = view.findViewById(R.id.textView2);
        TextView priceText = view.findViewById(R.id.textView3);
        TextView fulldescribtionText = view.findViewById(R.id.textView4);
        titleText.setText(name);
        describtionText.setText(describtion);
        priceText.setText(price +"$");
        fulldescribtionText.setText(fullDescribtion);
        return view;
    }
}

File: app/src/main/java/com/example/myapplication/DetailFragment.kt
package com.example.myapplication

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import android.widget.TextView
import com.example.myapplication.R

class DetailFragment : Fragment() {
    private lateinit var sport: OlympicSport

    companion object {
        fun newInstance(sport: OlympicSport): DetailFragment {
            val fragment = DetailFragment()
            fragment.sport = sport
            return fragment
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_detail, container, false)
        val nameTextView: TextView = view.findViewById(R.id.sportName)
        val typeTextView: TextView = view.findViewById(R.id.sportType)
        val yearTextView: TextView = view.findViewById(R.id.sportYear)

        nameTextView.text = sport.name
        typeTextView.text = if (sport.isTeam) "Командный" else "Одиночный"
        yearTextView.text = sport.yearRecognized.toString()

        return view
    }
}